src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -163,58 +158,58 @@
-                          boolean enableGcDeletes, long gcDeletesInMillis, ByteSizeValue indexingBufferSize, String codecName,
-                          boolean compoundOnFlush, int indexConcurrency, boolean optimizeAutoGenerateId, boolean failEngineOnCorruption,
-                          FailedEngineListener failedEngineListener) throws EngineException {
-        Preconditions.checkNotNull(store, "Store must be provided to the engine");
-        Preconditions.checkNotNull(deletionPolicy, "Snapshot deletion policy must be provided to the engine");
-        Preconditions.checkNotNull(translog, "Translog must be provided to the engine");
-
-        this.shardId = shardId;
-        this.logger = logger;
-        this.gcDeletesInMillis = gcDeletesInMillis;
-        this.enableGcDeletes = enableGcDeletes;
-        this.indexingBufferSize = indexingBufferSize;
-        this.codecName = codecName;
-
-        this.threadPool = threadPool;
+    public InternalEngine(EngineConfig config) throws EngineException {
+        Preconditions.checkNotNull(config.getStore(), "Store must be provided to the engine");
+        Preconditions.checkNotNull(config.getDeletionPolicy(), "Snapshot deletion policy must be provided to the engine");
+        Preconditions.checkNotNull(config.getTranslog(), "Translog must be provided to the engine");
+
+        this.shardId = config.getShardId();
+        this.logger = Loggers.getLogger(getClass(), config.getIndexSettings(), shardId);
+        this.threadPool = config.getThreadPool();
         this.lastDeleteVersionPruneTimeMSec = threadPool.estimatedTimeInMillis();
-        this.indexingService = indexingService;
-        this.warmer = (InternalIndicesWarmer) warmer;
-        this.store = store;
-        this.deletionPolicy = deletionPolicy;
-        this.translog = translog;
-        this.mergePolicyProvider = mergePolicyProvider;
-        this.mergeScheduler = mergeScheduler;
-        this.analysisService = analysisService;
-        this.similarityService = similarityService;
-        this.codecService = codecService;
-        this.compoundOnFlush = compoundOnFlush;
-        this.indexConcurrency = indexConcurrency;
+        this.indexingService = config.getIndexingService();
+        this.warmer = config.getWarmer();
+        this.store = config.getStore();
+        this.deletionPolicy = config.getDeletionPolicy();
+        this.translog = config.getTranslog();
+        this.mergePolicyProvider = config.getMergePolicyProvider();
+        this.mergeScheduler = config.getMergeScheduler();
+        this.analysisService = config.getAnalysisService();
+        this.similarityService = config.getSimilarityService();
+        this.codecService = config.getCodecService();
         this.versionMap = new LiveVersionMap();
-        this.dirtyLocks = new Object[indexConcurrency * 50]; // we multiply it to have enough...
+        this.dirtyLocks = new Object[config.getIndexConcurrency() * 50]; // we multiply it to have enough...
```

Comment:
Can the collapse the multiplying by 50 into the setting itself? It feels arbitrary and I have no idea where the 50 came from